test/test_helper.rb

```diff
@@ -34,8 +35,19 @@ def stub_warden_as(user)
     )
   end
 
+  def mock_env(partial_env_hash)
+    old = ENV.to_hash
+    ENV.update partial_env_hash
+    begin
+      yield
+    ensure
+      ENV.replace old
+    end
+  end
```

Comment:
Nice method! Did you reuse this from anywhere or is it new? I seem to remember seeing other alphagov repos use a ruby gem for handling temporary ENV vars, but I can't find the name of it now 🤔

Reply:
I stole it from here: https://gist.github.com/jazzytomato/79bb6ff516d93486df4e14169f4426af
I used https://github.com/thoughtbot/climate_control before but I wasn't sure about introducing another dependency but more than happy to use it.

Reply:
Ahhh yes, climate_control is what I was thinking of. But if this works, LGTM 👍